package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2032,6 +2045,239 @@ def total_charge(group, compound='group'):
+        accordingly.
+
+        Parameters
+        ----------
+        group : obj
+            AtomGroup
+        wrap : bool, optional
+            If ``True`` and `compound` is ``'group'``, move all atoms to the
+            primary unit cell before calculation.
+            If ``True`` and `compound` is not ``group``, the
+            centers of mass of each compound will be calculated without moving
+            any atoms to keep the compounds intact.
+        unwrap : bool, optional
+            If ``True``, compounds will be unwrapped before computing their
+            centers.
+        compound : {'group', 'segments', 'residues', 'molecules', \
+                    'fragments'}, optional
+            If ``'group'``, a single dipole vector returns. Otherwise, an
+            array of each :class:`Segment`, :class:`Residue`, molecule, or
+            fragment will be returned as an array of position vectors, i.e.
+            a 2d array.
+            Note that, in any case, *only* the positions of
+            :class:`Atoms<Atom>` *belonging to the group* will be taken into
+            account.
+        center : str, optional
+            Choose whether the dipole moment is calculated at the center of 
+            "mass" or the center of "charge"
+
+        Returns
+        -------
+        numpy.ndarray
+            Dipole moment of (compounds of) the group in e*angstroms.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,)``
+            where ``n`` is the number of compounds.
+
+        .. versionadded:: 2.4.0
+        """
+
+        compound = compound.lower()
+
+        atomgroup = group.atoms
+        charges = atomgroup.charges
+
+        if center == "mass":
+            masses = atomgroup.masses
+            ref = atomgroup.center_of_mass(
+                wrap=wrap, 
+                unwrap=unwrap, 
+                compound=compound
+            )
+        elif center == "charge":
+            ref = atomgroup.center_of_charge(
+                wrap=wrap,
+                unwrap=unwrap,
+                compound=compound
+            )
+        else:
+            choices = ["mass", "charge"]
+            raise ValueError(
+                "The dipole center, {}, is not supported. Choose: {}".format(
+                 center,choices))
+
+        if compound == 'group':
+            if wrap:
+                recenteredpos = (atomgroup.pack_into_box(inplace=False) - ref)
+            elif unwrap:
+                recenteredpos = (atomgroup.unwrap(inplace=False) - ref)
+            else:
+                recenteredpos = (atomgroup.positions - ref)
+            dipole_moment = np.sum(recenteredpos*charges[:,np.newaxis], axis=0)
+        else:
+            (atom_masks, 
+             compound_masks, 
+             n_compounds) = atomgroup._split_by_compound_indices(compound)
+
+            if unwrap:
```

Comment:
Does this not allow `wrap` as in the previous branch?

Reply:
That was an oversight on my part, thanks!

Reply:
IIRC this was also missed in Quadrupole, just thought I would mention before I forgot.